ddtrace/appsec/ddwaf/__init__.py

```diff
@@ -126,33 +139,34 @@ def run(
             # type: (...) -> DDWaf_result
             start = time.time()
 
-            ctx = ddwaf_context_init(self._handle)
-            if ctx == 0:
-                LOGGER.error("DDWaf failure to create the context")
+            if self._ctx == 0:
+                LOGGER.warning("DDWaf failsafe to create the context")
```

Comment:
is not this validation redundant with `_at_request_start`?

Reply:
It's a failsafe. Yes it's redondant but in the case that span_start was not called for any reason before the call to the WAF, it allows to process as usual without errors.